### PR TITLE
Fix Record query slow for species observations

### PR DESCRIPTION
**Pull request by serge-gaia**
The problem was that the `RecordSerializer` was loading a `name_id` -> `species_name` mapping from herbie every time if was used with a Species Observation dataset. This mapping was loaded even for a deserialisation (!) hence the delay for any get.
Now this mapping (used for name_id extraction/validation) should be used/loaded only for post/put/patch.

### CODE REVIEW

File: biosys/apps/main/tests/api/test_records_upload.py [hunk 1]
```diff
@@ -205,6 +205,36 @@ def test_upload_blank_column(self):
             self.assertEquals(self.project_1.record_count, len(csv_data) - 1)
             self.assertEquals(self.ds.record_count, len(csv_data) - 1)
 
+    def test_unicode(self):
```
Comment (serge-gaia) [inline biosys/apps/main/tests/api/test_records_upload.py:208]: This a missing test for the previous PR about unicode issue.
